xcm/xcm-builder/src/pay.rs

```diff
@@ -31,28 +31,20 @@ use xcm_executor::traits::{Convert, QueryResponseStatus, XcmQueryHandler};
 /// `PayOverXcm::pay` is asynchronous, and returns a `QueryId` which can then be used in
 /// `check_payment` to check the status of the XCM transaction.
 ///
-pub struct PayOverXcm<
-	DestChain,
-	Router,
-	Querier,
-	BlockNumber,
-	Timeout,
-	AccountId,
-	AccountIdConverter,
->(PhantomData<(DestChain, Router, Querier, BlockNumber, Timeout, AccountId, AccountIdConverter)>);
+pub struct PayOverXcm<Sender, Router, Querier, Timeout, Beneficiary, AssetKind>(
+	PhantomData<(Sender, Router, Querier, Timeout, Beneficiary, AssetKind)>,
+);
 impl<
-		DestChain: Get<xcm::v3::MultiLocation>,
+		Sender: Get<xcm::v3::MultiLocation>,
 		Router: SendXcm,
 		Querier: XcmQueryHandler,
-		BlockNumber,
 		Timeout: Get<Querier::BlockNumber>,
-		AccountId: Clone,
-		AccountIdConverter: Convert<MultiLocation, AccountId>,
-	> Pay
-	for PayOverXcm<DestChain, Router, Querier, BlockNumber, Timeout, AccountId, AccountIdConverter>
+		Beneficiary: Into<xcm::v3::MultiLocation> + Clone,
+		AssetKind: Into<xcm::v3::AssetId>,
+	> Pay for PayOverXcm<Sender, Router, Querier, Timeout, Beneficiary, AssetKind>
 {
-	type Beneficiary = AccountId;
-	type AssetKind = xcm::v3::AssetId;
+	type Beneficiary = Beneficiary;
+	type AssetKind = AssetKind;
```

Comment:
keeping these generic, since we wanna have a general solution here.
for treasury case can be, 
`enum Beneficiary {
    Local(AccountId),
    Statemint(AccountId)
}`
`enum AssetKind {
    LocalNative,
    Statemint(AssetId)
}`

where only `Beneficiary::Statemint(...)` routed to this `Pay` impl